test/forward/forward.jl

```diff
@@ -1,7 +1,16 @@
-using Zygote, Test
+using ChainRulesCore
+using Test
+using Zygote
 using NNlib: relu
 
-D(f, x) = pushforward(f, x)(1)
+foo(x) = x
+
+# This intentionally has the wrong definition so that we can detect if Zygote is using it.
+ChainRulesCore.frule((_, dx), ::typeof(foo), x) = x, 2 * dx
```

Comment:
the tests in the reverse mode chainrules tests set a variable that can late be read to see if it was hit.
This should be tested the same way

Reply:
Oh nice, I'll do that.